crates/router/src/connector/forte/transformers.rs

```diff
@@ -63,62 +66,93 @@ impl TryFrom<utils::CardIssuer> for ForteCardType {
     }
 }
 
-impl TryFrom<&types::PaymentsAuthorizeRouterData> for FortePaymentsRequest {
+#[derive(Debug, Serialize)]
+pub struct ForteRouterData<T> {
+    amount: i64,
+    router_data: T,
+}
+impl<T>
+    TryFrom<(
+        &types::api::CurrencyUnit,
+        types::storage::enums::Currency,
+        i64,
+        T,
+    )> for ForteRouterData<T>
+{
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        if item.request.currency != enums::Currency::USD {
-            Err(errors::ConnectorError::NotSupported {
-                message: item.request.currency.to_string(),
-                connector: "Forte",
-            })?
-        }
-        match item.request.payment_method_data {
-            api_models::payments::PaymentMethodData::Card(ref ccard) => {
-                let action = match item.request.is_auto_capture()? {
+    fn try_from(
+        (_currency_unit, _currency, amount, item): (
+            &types::api::CurrencyUnit,
+            types::storage::enums::Currency,
+            i64,
+            T,
+        ),
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount,
+            router_data: item,
+        })
+    }
```

Comment:
Convert the `amount` into `f64` using this function `utils::to_currency_base_unit_asf64`